apps/project-renderer/src/components/app/index.js

```diff
@@ -76,34 +80,32 @@ const App = ( {
 			form.append( key, data[ key ] )
 		);
 
-		return (
-			window
-				.fetch(
-					`https://api.crowdsignal.com/v4/projects/${ projectCode }/form`,
-					{
-						method: 'POST',
-						body: form,
-					}
-				)
-				.then( ( res ) => {
-					if ( ! res.ok ) {
-						throw new Error( res.status );
-					}
-
-					return res.json();
-				} )
-				.then( ( json ) => {
-					if ( ! json || ! json.content ) {
-						throw new Error( 'Empty response' );
-					}
-
-					setContent( json.content );
-					setHasResponded( true );
-					// all the setters should be called here: page, responseHash, content and startTime
-				} )
-				// eslint-disable-next-line no-console
-				.catch( ( err ) => console.error( err ) )
-		);
+		return window
+			.fetch(
+				`https://api.crowdsignal.com/v4/projects/${ projectCode }/form`,
+				{
+					method: 'POST',
+					body: form,
+				}
+			)
+			.then( ( res ) => {
+				if ( ! res.ok ) {
+					throw new Error( res.status );
+				}
+
+				return res.json();
+			} )
+			.then( ( json ) => {
+				if ( ! json || ! json.content ) {
+					throw new Error( 'Empty response' );
+				}
+
+				const responseWithErrors = ( json.errors || [] ).length > 0;
+				setHasErrors( responseWithErrors );
+				setContent( json.content );
```

Comment:
This way of doing things feels very obscure. Wouldn't it be better, if we just get `errors` ? Like:

```
{
    `errors: {
        `someClientId`: [
            `bla bla bla`,
             ...,
         ],
     },
}
```

And then we can manually dispatch `setFieldError` from here to push them straight into the form state?

Reply:
I'm probably guilty on this one. Christian and I had some discussions about this implementation and I suggested sending the errors to the `<Form />` and handling them from there.
Settings the field errors from the project-renderer would be much easier but I'm wondering if this is project-renderer's responsibility.
Since the `<Form /`> component already handles the fields validations, it seems more natural to me that it also handles those field errors.

Reply:
To some extend. In this case, we're not really talking about 'field errors' but rather form submission errors - which would best stay confined to the submit handler.  

To be completely honest, I still can't quite tell where the error messages are coming from or going to, or if they're used for anything at all other than setting `hasErrors`, just by reading the code.